File: 7-kyu/credit-card-issuer-checking/python/solution.py
def get_issuer(number):
    number = str(number)
    if number[:2] == '34' or number[:2] == '37' and len(number) == 15:
        return 'AMEX'
    elif number[:4] == '6011' and len(number) == 16:
        return 'Discover'
    elif 55 >= int(number[:2]) >= 50 and len(number) == 16:
        return 'Mastercard'
    elif number[0] == '4' and (len(number) == 13 or len(number) == 16):
        return 'VISA'
    else:
        return 'Unknown'

File: 7-kyu/credit-card-issuer-checking/c++/solution.cpp
std::string getIssuer(const std::string &number)
{
    if (number.substr(0, 2) == "34" || number.substr(0, 2) == "37" && number.length() == 15) return "AMEX";
    else if (number.substr(0, 4) == "6011" && number.length() == 16) return "Discover";
    else if (number[0] == '5' && number[1] - '0' <= 5) return "Mastercard";
    else if (number[0] == '4' && (number.length() == 16 || number.length() == 13)) return "VISA";
    else return "Unknown";
}